src/shared/components/data-table/components/main/components/table-body/TableBody.tsx

```diff
@@ -112,4 +100,49 @@ const TableBody = () => {
   );
 };
 
+type GetCurrentPageRowsParams = Pick<
+  DataTableState,
+  'hiddenRowIds' | 'data' | 'currentPageNumber' | 'rowsPerPage'
+> & {
+  uniqueColumnId: string;
+  columns: DataTableColumns;
+};
+
+export const getCurrentPageRows = (params: GetCurrentPageRowsParams) => {
+  const {
+    hiddenRowIds,
+    data,
+    currentPageNumber,
+    uniqueColumnId,
+    rowsPerPage,
+    columns
+  } = params;
+
+  const uniqueColumnIndex = columns.findIndex(
+    (column) => column.columnId === uniqueColumnId
+  );
+
+  const totalRows = data.length;
+  const rowIndexLowerBound = (currentPageNumber - 1) * rowsPerPage;
+  const rowIndexUpperBound = rowIndexLowerBound + rowsPerPage;
+
+  const visibleRows = hiddenRowIds
+    ? data.filter((rowData) => {
+        const rowId = String(rowData[uniqueColumnIndex]);
+
+        return hiddenRowIds[rowId] !== true;
```

Comment:
Isn't this the same as `return !hiddenRowIds[rowId]`?

Reply:
That's done